File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/VisualBasic/Extensibility/1033/Margin/MarginFactory.vb
Imports System.ComponentModel.Composition
Imports Microsoft.VisualStudio.Text.Editor
Imports Microsoft.VisualStudio.Utilities

''' <summary>
''' Export a <see cref="IWpfTextViewMarginProvider"/>, which returns an instance of the margin for the editor
''' to use.
''' </summary>
<Export(GetType(IWpfTextViewMarginProvider))>
<Name($safeprojectname$.MarginName)>
<Order(After := PredefinedMarginNames.HorizontalScrollBar)>
<MarginContainer(PredefinedMarginNames.Bottom)>
<ContentType("text")>
<TextViewRole(PredefinedTextViewRoles.Interactive)> 
NotInheritable Class MarginFactory
    Implements IWpfTextViewMarginProvider
    
    Public Function CreateMargin(ByVal textViewHost As IWpfTextViewHost, ByVal containerMargin As IWpfTextViewMargin) As IWpfTextViewMargin Implements IWpfTextViewMarginProvider.CreateMargin
        Return New $safeprojectname$(textViewHost.TextView)
    End Function
    
End Class

File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/CSharp/Extensibility/1033/Margin/MarginFactory.cs
﻿using System.ComponentModel.Composition;
using Microsoft.VisualStudio.Text.Editor;
using Microsoft.VisualStudio.Utilities;

namespace $safeprojectname$
{
    #region $safeprojectname$ Factory
    /// <summary>
    /// Export a <see cref="IWpfTextViewMarginProvider"/>, which returns an instance of the margin for the editor
    /// to use.
    /// </summary>
    [Export(typeof(IWpfTextViewMarginProvider))]
    [Name($safeprojectname$.MarginName)]
    [Order(After = PredefinedMarginNames.HorizontalScrollBar)] //Ensure that the margin occurs below the horizontal scrollbar
    [MarginContainer(PredefinedMarginNames.Bottom)] //Set the container to the bottom of the editor window
    [ContentType("text")] //Show this margin for all text-based types
    [TextViewRole(PredefinedTextViewRoles.Interactive)]
    internal sealed class MarginFactory : IWpfTextViewMarginProvider
    {
        public IWpfTextViewMargin CreateMargin(IWpfTextViewHost textViewHost, IWpfTextViewMargin containerMargin)
        {
            return new $safeprojectname$(textViewHost.TextView);
        }
    }
    #endregion
}
